### PR TITLE
Fix validation and outdated schema problem

### PR DESCRIPTION
**Pull request by gnawf**

### CODE REVIEW

File: lib/src/main/java/graphql/nadel/schema/NadelDirectives.kt [hunk 1]
```diff
@@ -223,7 +223,10 @@ object NadelDirectives {
                 val identifiedByValues = resolveArgumentValue<List<Any>>(inputIdentifiedBy)
                 val identifiedBy = createObjectIdentifiers(identifiedByValues)
 
-                val conditionalHydration = buildConditionalHydrationObject(directive.getArgument("when"))?.result
+                val conditionalHydration = directive.getArgument("when")
+                    ?.let {
```
Comment (gnawf) [inline lib/src/main/java/graphql/nadel/schema/NadelDirectives.kt:227]: Handle case where schema does not define `when` and the value is null.

File: lib/src/main/java/graphql/nadel/validation/NadelHydrationWhenConditionValidation.kt [hunk 2]
```diff
@@ -26,17 +25,19 @@ internal class NadelHydrationWhenConditionValidation() {
...
+        val whenConditionSourceFieldType = whenConditionSourceField.type.unwrapNonNull()
+        if (whenConditionSourceFieldType !is GraphQLScalarType) {
             return NadelSchemaValidationError.WhenConditionUnsupportedFieldType(
                 whenConditionSourceFieldName,
                 GraphQLTypeUtil.simplePrint(whenConditionSourceField.type),
-                overallField
+                overallField,
             )
         }
-        val whenConditionSourceFieldTypeName: String = (whenConditionSourceField.type as GraphQLScalarType).name
```
Comment (gnawf) [inline lib/src/main/java/graphql/nadel/validation/NadelHydrationWhenConditionValidation.kt:39]: Avoid manual casting. There is a bug where a non-nullable type causes this to crash.

Comment (sbarker2) [reply lib/src/main/java/graphql/nadel/validation/NadelHydrationWhenConditionValidation.kt:39]: Thanks for fixing this 👍

File: lib/src/main/java/graphql/nadel/schema/NadelDirectives.kt (hunk 1 above)
Comment (sbarker2) [reply lib/src/main/java/graphql/nadel/schema/NadelDirectives.kt:227]: makes sense, thanks for picking it up 👍

**Review by sbarker2 — APPROVED** [review]

**Review by sbarker2 — APPROVED** [review]

**Pull request merged**